la_funding_analysis/utils/name_cleaners.py

```diff
@@ -0,0 +1,77 @@
+# File: utils/name_cleaners.py
+"""Functions to clean local authority names and model codes
+in order to join data from difference sources
+and make plots more readable.
+"""
+
+import numpy as np
+import pandas as pd
+
+
+def clean_names(name) -> str:
+    # Function to clean local authority names in various datasets
+    # in order to join data from different sources.
+    strings = [
+        " Metropolitan Borough Council",
+        " Metropolitan District Council",
+        " Royal Borough Council",
+        "Royal Borough of ",
+        "London Borough of ",
+        " Borough Council",
+        " District Council",
+        " City Council",
+        " County Council",
+        "District",
+        "Council",
+        "Corporation",
+        ", City of",
+        "City of ",
+        ", County of",
+        "County UA",
+        "County ",
+        " (Met County)",
+        "CC",
+        "DC",
+    ]
+    name = name.replace("\xa0", " ")
+    for string in strings:
+        name = name.replace(string, "")
+    name = (
+        name.replace("&", "and")
+        .replace("Mid-", "Mid ")
+        .replace("Upon", "upon")
+        .replace("Kings Lynn", "King's Lynn")
+        .replace("King’s Lynn", "King's Lynn")
+        .replace("Basingstoke and Deane", "Basingstoke and Dean")
+        .replace("St Helens", "St. Helens")
+        .replace("Vale of Whitehorse", "Vale of White Horse")
+        .replace("Newcastle upon Tyne", "Newcastle")
+        .strip()
+    )
```

Comment:
There's nothing strictly wrong with this, but it might be neater to iterate through key value pairs in a dictionary and call `.replace()` in a loop

Reply:
Actioned